hmpy/views/gauge.py

```diff
@@ -0,0 +1,289 @@
+from hmpy.views.view import View
+from PyQt5.QtGui import QPainter, QBrush, QPen, QColor, QImage, QConicalGradient, QPalette, QPixmap, QPolygon, QFont, QFontMetrics
+from PyQt5.QtCore import Qt, QRect, QPoint, QPointF
+from enum import Enum
+import math
+
+
+class GaugeView(View):
+    """
+    View simulating a Gauge.
+    """
+
+    def __init__(self, value=0, min_value=0, max_value=100, precision=2,
+                 unit_text="", color=QColor(Qt.white)):
+        """Initialize the view
+
+        :param starting_value: starting value for the gauge.
+        :param min_value: The minimum value of the gauge.
+        :param max_value: The minimum value of the gauge.
+        :param precision: The minimum value of the gauge.
+        :param unit_text: The text displaying the unit of the gauge
+        :param color: The background colour of the gauge
+        """
+        super().__init__()
+
+        self._color = color
+        self._unit_text = unit_text
+        self.__value = value
+
+        self.__min_value = min_value
+        self.__max_value = max_value
+
+        self._precision = precision
+
+        self._pointText = []
+        self._create_point_text()
+        self._needle_angle = -135
+        self._calculate_needle_angle()
+
+
+    @property
+    def value(self):
+        """Get 'value' class attribute.
+
+        :return: the value of the 'value' attribute.
+        """
+        return self.__value
+
+    @value.setter
+    def value(self, value):
+        """Set 'value' class attribute, trigger repaint.
+
+        :param to: Boolean value.
+        """
+        if value < self.__min_value:
```

Comment:
I think limiting the needle is a good option.

Reply:
So we would probably want to perform this check only when rendering the needle.  How often is this running?  Would this happen only on value change or does something else trigger the render process?  If it changes alot it could potentially be more efficient to do the check upfront and store it in a separate class member